pkg/agent/wireguard/client_linux.go

```diff
@@ -256,3 +271,15 @@ func (client *client) DeletePeer(nodeName string) error {
 	client.peerPublicKeyByNodeName.Delete(nodeName)
 	return nil
 }
+
+func (client *client) Cleanup() error {
+	_, err := netlink.LinkByName(client.wireGuardConfig.Name)
+	if err != nil {
```

Comment:
If `LinkByName` tries to get a non-existent interface, it will return an error with message "xxx not found". But if we use `LinkDel` to get a  non-existent interface, it will just return "invalid argument", so maybe we cannot remove the pre-check?

Reply:
> I tried but I found that we cannot use a function just like LinkExists. The pkg/agent/util/netlink is just an interface implemented by the netlink package directly. So the interface must be a subset of the netlink interface. So maybe we cannot use LinkExists.

I do not mean to add the func to the `netlink` interface, but as a standalone func in netlink_linux.go

Reply:
> If `LinkByName` tries to get a non-existent interface, it will return an error with message "xxx not found". But if we use `LinkDel` to get a non-existent interface, it will just return "invalid argument", so maybe we cannot remove the pre-check?

I guess it returns "invalid argument" because you didn't set `link.Index`. And `client` already caches the interface index, I think we can pass it? Then the pre-check is not needed.

Reply:
> I guess it returns "invalid argument" because you didn't set link.Index. And client already caches the interface index, I think we can pass it? Then the pre-check is not needed.

Hi Quan. I tried to pass the link.Index to the LinkDel function, it returned no such device. Now I'm not sure which one is better, just using LinkDel and check the error message with strings.Contain or using LinkByName first and checking the error type (The netlink package wraps "not found" error when in LinkByName, but not in Linkdel).

Reply:
I think you could check "no such device", this is not the first place we check error message to confirm if the resource has been deleted: https://github.com/antrea-io/antrea/blob/d2c4ef86b6aab3118316b7f51c0bad1753ff7495/pkg/agent/route/route_linux.go#L1390

Reply:
Done, removed the `LinkExists` and add a check for the error message.